src/module/Provider.sol

```diff
@@ -12,61 +21,47 @@ import {MainnetContracts as MC} from "script/Contracts.sol";
 contract Provider is IProvider {
     error UnsupportedAsset(address asset);
 
-    mapping(address => uint256) public rates; // in WETH
-
     function getRate(address asset) external view override returns (uint256) {
         if (asset == MC.WETH) {
             return 1e18;
-        } else if (asset == MC.STETH) {
+        }
+
+        if (asset == MC.STETH) {
             return 1e18;
-        } else if (asset == MC.YNETH) {
-            return IERC4626(MC.YNETH).previewRedeem(1e18);
-        } else if (asset == MC.YNLSDE) {
-            return IERC4626(MC.YNLSDE).previewRedeem(1e18);
-        } else if (asset == MC.BUFFER) {
-            return IERC4626(MC.BUFFER).previewRedeem(1e18);
-        } else if (asset == MC.WSTETH) {
-            return IStETH(MC.STETH).getPooledEthByShares(1e18);
-        } else if (asset == MC.METH) {
-            return IMETH(MC.METH).ratio();
-        } else if (asset == MC.OETH) {
-            return IOETH(MC.OETH).assetToEth(1e18);
-        } else if (asset == MC.RETH) {
-            return IRETH(MC.RETH).getExchangeRate();
         }
 
-        revert UnsupportedAsset(asset);
-    }
-}
+        if (asset == MC.BUFFER || asset == MC.YNETH || asset == MC.YNLSDE || asset == MC.WOETH) {
+            return IERC4626(asset).previewRedeem(1e18);
+        }
 
-interface IStETH {
-    function getPooledEthByShares(uint256 _ethAmount) external view returns (uint256);
-}
+        if (asset == MC.WSTETH) {
+            return IStETH(MC.STETH).getPooledEthByShares(1e18);
+        }
 
-interface IMETH {
-    function ratio() external view returns (uint256);
-}
+        if (asset == MC.METH) {
+            return IMETH(MC.METH_STAKING_MANAGER).mETHToETH(1e18);
```

Comment:
noticed the change here, this looks correct, i had never used the previous one before with ratio()